wom/src/main/scala/wom/types/WomFloatType.scala

```diff
@@ -11,8 +11,11 @@ case object WomFloatType extends WomPrimitiveType {
   override protected def coercion = {
     case f: Float => WomFloat(f.toDouble)
     case d: Double => WomFloat(d)
+    case i: Integer => WomFloat(i.toDouble)
     case n: JsNumber => WomFloat(n.value.doubleValue())
     case f: WomFloat => f
+    case i: WomInteger => WomFloat(i.value.toDouble)
+    case l: WomLong => WomFloat(l.value.toDouble)
```

Comment:
I personally love this, but I think the above breaks the WDL draft-2 spec as ints cannot be cast to floats. (And it doesn't even mention "long" so...)

https://github.com/openwdl/wdl/blob/master/versions/draft-2/SPEC.md#type-coercion

cc: @cjllanwarne @aednichols for their thoughts.